source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -1000,24 +1007,38 @@ TEST(NiFakeServiceTests, NiFakeService_MultipleArraysSameSize_OneArrayWithZeroSi
   EXPECT_EQ(::grpc::INVALID_ARGUMENT, status.error_code());
 }
 
+FakeCustomStruct create_custom_struct(pb::int32 struct_int, double struct_double)
+{
+  auto custom_struct = FakeCustomStruct{};
+  custom_struct.set_struct_int(struct_int);
+  custom_struct.set_struct_double(struct_double);
+  return custom_struct;
+}
+
 TEST(NiFakeServiceTests, NiFakeService_MultipleArraysSameSizeWithOptionals_OneArrayWithZeroSizePassesNull)
 {
   nidevice_grpc::SessionRepository session_repository;
   NiFakeMockLibrary library;
   auto resource_repository = std::make_shared<FakeResourceRepository>(&session_repository);
   nifake_grpc::NiFakeService service(&library, resource_repository);
   auto session_id = create_session(library, service, kTestViSession);
-  const std::vector<double> doubles = {0.2, -2.3, 4.5};
+  const auto doubles = std::vector<double>{0.2, -2.3, 4.5};
+  const auto fake_structs = std::vector<FakeCustomStruct>{
+      create_custom_struct(1, 3.2),
+      create_custom_struct(0, 10.0),
+      create_custom_struct(100, 0.)};
```

Comment:
observation: ive never seen "0." obviously it is compiling as a double, but I would expected 0.0

Reply:
Maybe that's too cute. 

From [cppreference ](https://en.cppreference.com/w/c/language/floating_constant):

> For decimal floating constants, the exponent part is optional. If it is omitted, the period is not optional, and either the whole-number or the fraction must be present.

Reply:
I'm going to keep it to keep the PRs moving.